addons/binding/org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/rio/AbstractBridgeHandler.java

```diff
@@ -77,6 +98,38 @@ public SocketSession getSocketSession() {
     }
 
     /**
+     * Returns the {@link RioPresetsProtocol} for this {@link Bridge}. The default implementation is to look in the
+     * parent
+     * {@link #getPresetsProtocol()} for the {@link RioPresetsProtocol}
+     *
+     * @return a {@link RioPresetsProtocol} or null if none exists
+     */
+    @SuppressWarnings("rawtypes")
+    public RioPresetsProtocol getPresetsProtocol() {
+        final Bridge bridge = getBridge();
+        if (bridge.getHandler() instanceof AbstractBridgeHandler) {
+            return ((AbstractBridgeHandler) bridge.getHandler()).getPresetsProtocol();
+        }
+        return null;
+    }
+
+    /**
+     * Returns the {@link RioSystemFavoritesProtocol} for this {@link Bridge}. The default implementation is to look in the
+     * parent
+     * {@link #getPresetsProtocol()} for the {@link RioSystemFavoritesProtocol}
+     *
+     * @return a {@link RioSystemFavoritesProtocol} or null if none exists
+     */
+    @SuppressWarnings("rawtypes")
+    public RioSystemFavoritesProtocol getSystemFavoritesHandler() {
+        final Bridge bridge = getBridge();
+        if (bridge != null && bridge.getHandler() instanceof AbstractBridgeHandler) {
```

Comment:
I would expect that since line 109 and 125 are the same and line 111 and 127 are doing more or less the same (calling a method on the bridge its handler). I would expected the lines 110 and 125 to be resilient to the same amount of `null` values. But I might miss some knowledge which would make it unnecessary.